### PR TITLE
Ch19176/add mfa otp grant type to oauth package

### PR DESCRIPTION
**Pull request by andrewrod33**
Add step 2 of MFA Login

- [x] Test on Staging
- [x] Make sure expires_at works
- [x] Make sure expires_in works
- [x] Make sure scope works

### CODE REVIEW

File: lib/grant.js [hunk 1]
```diff
@@ -313,6 +315,48 @@ function useExtendedGrant (done) {
...
 
+/**
+ * Grant for urn:custom:mfa-otp (http://*) grant type
+ *
+ * @param  {Function} done
+ */
+function useMfaOtpGrant (done) {
+  var self = this;
+  if (self.req.body) {
+    if (!self.req.body.otp || !self.req.body.mfa_token) {
```
Comment (monkbroc) [inline lib/grant.js]: Can be simplified with
```
    if (!self.req.body || !self.req.body.otp || !self.req.body.mfa_token) {
```

File: test/grant.mfaOtp.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright 2013-present NightWorld.
```
Comment (monkbroc) [inline test/grant.mfaOtp.js]: They didn't write this code, so let's at least not put their copyright in this file :)

**Review by monkbroc — APPROVED** [review]
LGTM!

File: lib/grant.js (hunk 1 above)
Comment (andrewrod33) [reply lib/grant.js]: Fixed, thanks!

File: test/grant.mfaOtp.js (hunk 2 above)
Comment (andrewrod33) [reply test/grant.mfaOtp.js]: lol! Fixed, thanks!

File: lib/grant.js [hunk 3]
```diff
@@ -345,7 +345,7 @@ function useMfaOtpGrant (done) {
           self.req.body.expires_in = user.expires_in
       }
 
-      if (self.req.body && user.expires_at) {
+      if (self.req.body && user.expires_at !== undefined) {
```
Comment (monkbroc) [inline lib/grant.js]: I realize that this is probably an artifact of how the library works, but seeing `user.expires_at` looks really weird.

Comment (andrewrod33) [reply lib/grant.js]: Agreed.

**Pull request merged**